packages/framer-motion/src/animation/generators/spring/index.ts

```diff
@@ -0,0 +1,176 @@
+                        dampingRatio * undampedAngularFreq * initialDelta) /
+                        angularFreq) *
+                        Math.sin(angularFreq * t) +
+                        initialDelta * Math.cos(angularFreq * t))
+            )
+        }
+    } else if (dampingRatio === 1) {
+        // Critically damped spring
+        resolveSpring = (t: number) =>
+            target -
+            Math.exp(-undampedAngularFreq * t) *
+                (initialDelta +
+                    (initialVelocity + undampedAngularFreq * initialDelta) * t)
+    } else {
+        // Overdamped spring
+        const dampedAngularFreq =
+            undampedAngularFreq * Math.sqrt(dampingRatio * dampingRatio - 1)
+
+        resolveSpring = (t: number) => {
+            const envelope = Math.exp(-dampingRatio * undampedAngularFreq * t)
+
+            // When performing sinh or cosh values can hit Infinity so we cap them here
+            const freqForT = Math.min(dampedAngularFreq * t, 300)
+
+            return (
+                target -
+                (envelope *
+                    ((initialVelocity +
+                        dampingRatio * undampedAngularFreq * initialDelta) *
+                        Math.sinh(freqForT) +
+                        dampedAngularFreq *
+                            initialDelta *
+                            Math.cosh(freqForT))) /
+                    dampedAngularFreq
+            )
+        }
+    }
+
+    return {
+        calculatedDuration: isResolvedFromDuration ? duration || null : null,
+        next: (t: number) => {
+            const current = resolveSpring(t)
+
+            if (!isResolvedFromDuration) {
+                let currentVelocity = initialVelocity
+
+                if (t !== 0) {
+                    /**
+                     * We only need to calculate velocity for under-damped springs
+                     * as over- and critically-damped springs can't overshoot, so
+                     * checking only for displacement is enough.
+                     */
+                    if (dampingRatio < 1) {
+                        currentVelocity = calcGeneratorVelocity(
+                            resolveSpring,
+                            t,
+                            current
+                        )
+                    } else {
+                        currentVelocity = 0
+                    }
+                }
+
+                const isBelowVelocityThreshold =
+                    Math.abs(currentVelocity) <= restSpeed!
+                const isBelowDisplacementThreshold =
+                    Math.abs(target - current) <= restDelta!
+
+                state.done =
+                    isBelowVelocityThreshold && isBelowDisplacementThreshold
+            } else {
+                state.done = t >= duration!
+            }
+
+            state.value = state.done ? target : current
+
+            return state
+        },
+    }
+}
+
+const zero = (_t: number) => 0
```

Comment:
Why define this as a separate const? Performance?

Reply:
Basically yeah - this file is identical to the old `spring` and this is just a holdover. But I've just removed this and replaced with `let resolveSpring: (v: number) => number`